File: core/graph/graph_consistency_engine.py
from __future__ import annotations

from typing import Any, Dict

from core.graph.graph_invariant_engine import check_graph_invariants


def assess_graph_consistency(graph: Dict[str, Any]) -> Dict[str, Any]:
    inv = check_graph_invariants(graph)
    return {"consistent": inv["valid"], "invariants": inv, "deterministic_inputs": inv["deterministic_inputs"]}

File: src/graph/graphConsistencyEngine.ts
/**
 * Converted from Python: core/graph/graph_consistency_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { checkGraphInvariants } from "./graphInvariantEngine.js";

export function assessGraphConsistency(graph: any): any {
  var inv: any = checkGraphInvariants(graph);
  return {"consistent": py.at(inv, "valid"), "invariants": inv, "deterministic_inputs": py.at(inv, "deterministic_inputs")};
}
export { checkGraphInvariants };
